src/core/thread/mesh_forwarder.cpp

```diff
@@ -652,6 +653,21 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
                 VerifyOrExit(mMle.IsRouterIdValid(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
                 mMeshDest = rloc16;
             }
+            else if (mMle.IsAnycastLocator(ip6Header.GetDestination()))
+            {
+                // only support Leader ALOC for now
+                aloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
+
+                if (aloc16 == 0xfc00)
+                {
+                    mMeshDest = mMle.GetRloc16(mMle.GetLeaderId());
+                }
+                else
+                {
```

Comment:
I thing the logic here is for the on-mesh nodes (Router/Leader) to map ALOC to RLOC, while different roles ALOCs have different mapping rules, so we need if/else here. Do you agree?